packages/govuk-frontend/src/govuk/components/password-input/password-input.puppeteer.test.js

```diff
@@ -0,0 +1,363 @@
+          await page.click('[type="submit"]')
+
+          // Check the input type again
+          const afterSubmitType = await page.evaluate(
+            (inputClass) =>
+              document.querySelector(inputClass).getAttribute('type'),
+            inputClass
+          )
+
+          expect(beforeSubmitType).toEqual('text')
+          expect(afterSubmitType).toEqual('password')
+        })
+      })
+
+      describe('errors at instantiation', () => {
+        it('can throw a SupportError if appropriate', async () => {
+          await expect(
+            render(page, 'password-input', examples.default, {
+              beforeInitialisation() {
+                document.body.classList.remove('govuk-frontend-supported')
+              }
+            })
+          ).rejects.toMatchObject({
+            cause: {
+              name: 'SupportError',
+              message:
+                'GOV.UK Frontend initialised without `<body class="govuk-frontend-supported">` from template `<script>` snippet'
+            }
+          })
+        })
+
+        it('throws when $module is not set', async () => {
+          await expect(
+            render(page, 'password-input', examples.default, {
+              beforeInitialisation($module) {
+                $module.remove()
+              }
+            })
+          ).rejects.toMatchObject({
+            cause: {
+              name: 'ElementError',
+              message: 'Password input: Root element (`$module`) not found'
+            }
+          })
+        })
+
+        it('throws when receiving the wrong type for $module', async () => {
+          await expect(
+            render(page, 'password-input', examples.default, {
+              beforeInitialisation($module) {
+                // Replace with an `<svg>` element which is not an `HTMLElement` in the DOM (but an `SVGElement`)
+                $module.outerHTML = `<svg data-module="govuk-password-input"></svg>`
+              }
+            })
+          ).rejects.toMatchObject({
+            cause: {
+              name: 'ElementError',
+              message:
+                'Password input: Root element (`$module`) is not of type HTMLElement'
+            }
+          })
+        })
+
+        it('throws when the input element is missing', async () => {
+          await expect(
+            render(page, 'password-input', examples.default, {
+              beforeInitialisation($module, { selector }) {
+                $module.querySelector(selector).remove()
+              },
+              context: {
+                selector: inputClass
+              }
+            })
+          ).rejects.toMatchObject({
+            cause: {
+              name: 'ElementError',
+              message:
+                'Password input: Form field (`.govuk-js-password-input-input`) not found'
+            }
+          })
+        })
+
+        it('throws when the input element is not the right type', async () => {
```

Comment:
**question** This test checking against what class the element is an instance of made me wonder if it should also check if the `type` attribute of the input is appropriate (ie. `password`, or at least not something weird like `button`, `reset` or `image`). It might:
- have already been discussed (in which case, do ignore me)
- be a bit over the top (in which case, same)